File: Demo_pageLinkage/PageViewController.m
//
//  PageViewController.m
//  Demo_pageLinkage
//
//  Created by goulela on 2017/8/22.
//  Copyright © 2017年 MC. All rights reserved.
//

#import "PageViewController.h"

#import "SubViewController.h"


@interface PageViewController ()

@end

@implementation PageViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"页面联动";
    
    
    NSArray * dataArray = @[@"关注",@"推荐",@"热点",@"上海",@"娱乐",@"头条",@"问答",@"科技",@"视频",@"关注",@"推荐",@"热点",@"上海",@"娱乐",@"头条",@"问答",@"科技",@"视频"];
  
    
    //    NSArray * dataArray = @[@"关注",@"推荐"];
    //    self.barHeight = 40;
    //    self.isLeftSideDistribution = true;
    //    self.isHiddenBlock = true;
    
    
    NSMutableArray * vcArrayM = [NSMutableArray arrayWithCapacity:0];
    for (int i = 0; i < dataArray.count; i ++) {
        SubViewController * one = [[SubViewController alloc] init];
        one.title = dataArray[i];
        one.str = dataArray[i];
        [vcArrayM addObject:one];
        
        // 子页面上点击事件的处理  --> push到下个页面
        __weak __typeof__(self) weakSelf = self;
        one.oneBlock = ^(NSString *string, UIViewController *vc) {
            vc.title = string;
            [weakSelf.navigationController pushViewController:vc animated:YES];
        };
        
        // 子页面上点击事件的处理  --> 跳转到其他pageViewController的子页面
        one.twoBlock = ^(int index) {
            [weakSelf jumpToSubViewController:index];
        };
        
    }

    [self initWithTitleArray:dataArray vcArray:vcArrayM blockNormalColor:[UIColor lightGrayColor] blockSelectedColor:[UIColor redColor]];
    
    
    [self jumpToSubViewController:2];
}

@end

File: Example/MCPageViewController/PageViewController.swift
//
//  PageViewController.swift
//  MCPageViewController
//
//  Created by MC on 2018/11/5.
//  Copyright © 2018年 MC. All rights reserved.
//

import UIKit
import MCPageViewController

class PageViewController: MCPageViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "页面联动";
        let titles = ["第1页","第2页","第3页","第4页","第5页","第6页","第7页","第8页","第9页","第10页"]

//        let titles = ["第1页","第2页"]

        
        let vcArrayM = NSMutableArray()
        
        let arrayM = NSMutableArray()
        
        
        for i in 0..<titles.count {
            let vc = SubViewController()
            vc.delegate = self
            vc.str = titles[i]
            vcArrayM.add(vc)
            
            let item = MCPageItem()
            
            if i == 1 {
                let gif = "http://img.glela.com/file/base/20181116161837105.gif"
                
                item.tagImageView.sd_setImage(with: URL.init(string: gif), placeholderImage: nil)
                item.hotSize = CGSize.init(width: 12, height: 20)
//                item.tagImageView.image = #imageLiteral(resourceName: "hot")
                item.bgImageView.image = #imageLiteral(resourceName: "HOT_bg")
            }

            arrayM.add(item)
        }
        
        let config = MCPageConfig.init()
        config.titles = titles
        config.vcs = vcArrayM as! [UIViewController]
        config.blockWidth = 80
        config.indicatorColor = UIColor.orange
//        config.isLeftPosition = true
        
        
        initCustomPageWithConfig(config, items: arrayM as! [MCPageItem])
//        initPagesWithConfig(config)
    }
    

}

extension PageViewController : SubViewControllerProtocal {
    func jump(index: Int) {
        self.jumpToSubViewController(index)
    }
    
    func push() {
        let vc = NextViewController()
        self.navigationController?.pushViewController(vc, animated: true)
    }
}
